### PR TITLE
Add Prometheus labels

### PR DESCRIPTION
**Pull request by ezh**
Hi guys. You created a nice extension. Thanks.

I added custom labels to test metrics.

In the console with the `K6_DEBUG` it looks like
```
          /\      |‾‾| /‾‾/   /‾‾/
     /\  /  \     |  |/  /   /  /
    /  \/    \    |     (   /   ‾‾\
   /          \   |  |\  \ |  (‾)  |
  / __________ \  |__| \__\ \_____/ .io

Pushgateway labels from JSON options.ext.pushgateway dictionary map[app:myapp env:myenv]
Pushgateway labels map[app:myapp env:myenv]
  execution: local
```

I added a simple test as well and updated `README.md`

### CODE REVIEW

**Comment by ezh** [comment]
@martymarron WDYT?

**Comment by ezh** [comment]
BTW I saw different individuals as committers and authors, and I thought that you were some kind of a small company. 🤦 😆

**Comment by martymarron** [comment]
@ezh
Hi, thank you for suggesting this PR, and sorry for my belated response!
Because I'm pretty busy recently, I couldn't take the time for this.
I'll try to review this soon.
Thank you for your patience.

> BTW I saw different individuals as committers and authors, and I thought that you were some kind of a small company. 

Right, I've built this extension on several dev environments, and I forgot to unify Git usernames among them. Actually, all committer names are pointers that point to me 😅

**Review by martymarron — COMMENTED** [review]
\- (Comment deleted)

File: pkg/pushgateway/config.go [hunk 1]
```diff
@@ -15,17 +17,44 @@ const (
...
-		JobName:      defaultJobName,
+	}
+
+	if _, ok := p.Environment["K6_DEBUG"]; ok {
+		cfg.Debug = true
+	}
+
+	if val, ok := p.ScriptOptions.External["pushgateway"]; ok {
+		err := json.Unmarshal(val, &cfg.Labels)
+		if err != nil {
```
Comment (martymarron) [inline pkg/pushgateway/config.go:39]: Could you consider returning an error (`fmt.Errorf("...")`) instead?

File: pkg/pushgateway/config.go [hunk 2]
```diff
@@ -41,6 +70,13 @@ func NewConfig(p output.Params) (Config, error) {
 		case "K6_JOB_NAME":
 			cfg.JobName = v
 		}
+		if strings.HasPrefix(k, "K6_LABEL_") {
+			key := strings.ToLower(k[9:])
+			cfg.Labels[key] = strings.ToLower(v)
+		}
+	}
+	if cfg.Debug {
+		fmt.Printf("Pushgateway labels %+v\n", cfg.Labels)
```
Comment (martymarron) [inline pkg/pushgateway/config.go]: What do you think if this debug print is done at `Output#Description` as below?

```diff
func (o *Output) Description() string {
-	return "pushgateway: " + o.config.PushGWUrl
+	return fmt.Sprintf("pushgateway: %s, labels: %v", o.config.PushGWUrl. o.config.Labels)
}
```

**Review by martymarron — COMMENTED** [review]
@ezh 
Thank you for waiting!
I would appreciate your quick look at my comments 😉

**Comment by martymarron** [comment]
> In the console with the `K6_DEBUG` it looks like
> 
> ```
>           /\      |‾‾| /‾‾/   /‾‾/
>      /\  /  \     |  |/  /   /  /
>     /  \/    \    |     (   /   ‾‾\
>    /          \   |  |\  \ |  (‾)  |
>   / __________ \  |__| \__\ \_____/ .io
> 
> Pushgateway labels from JSON options.ext.pushgateway dictionary map[app:myapp env:myenv]
> Pushgateway labels map[app:myapp env:myenv]
>   execution: local
> ```

I'm considering if we can utilize `k6 run --verbose` instead of `K6_DEBUG` flag to print given labels for debugging purposes 🤔 
What do you think?

**Comment by ezh** [comment]
Thank you for your input. I read the comments carefully. I'm busy as well. But I'll reserve a time slot for that MR.
I agree with the list 😁

**Comment by martymarron** [comment]
@ezh 
Hi, thank you again for your contribution. I have something to ask.
I'm now creating the PR below.
[#2 Limit to push metrics single point in time per flush](https://github.com/martymarron/xk6-output-prometheus-pushgateway/pull/2)

Though I want to merge my PR by next Monday, my PR above contains conflicts w/ this PR.
Then, I shall merge my PR first unless you finish working on this PR by then.

Does my suggestion work for you?
Would really appreciate your feedback on this 😉

**Comment by ezh** [comment]
@martymarron Please move on. IMO it is okay to have conflicts, clean history, and do other housekeeping stuff. I'll make sure to update the code accordingly.

File: pkg/pushgateway/config.go (hunk 1 above)
Comment (ezh) [reply pkg/pushgateway/config.go:39]: I use `errors.Wrap`. The reason:
* it contains stack trace
* it is used by upstream libraries

File: pkg/pushgateway/config.go (hunk 2 above)
Comment (ezh) [reply pkg/pushgateway/config.go]: Done. It is much better with detailed Description output 🙏

**Comment by ezh** [comment]
Please, merge https://github.com/martymarron/xk6-output-prometheus-pushgateway/pull/2 first. I'll rebase on master shortly. It will be easy.

**Comment by martymarron** [comment]
> Please, merge #2 first. I'll rebase on master shortly. It will be easy.

Thank you for waiting. Merged!

**Comment by ezh** [comment]
@martymarron PR is rebased on the master branch.

**Comment by martymarron** [comment]
Will have a quick look soon. Thank you for your patience.

**Review by martymarron — APPROVED** [review]
LGTM

**Comment by martymarron** [comment]
@ezh 
Merged 🚀  Thank you for your contribution!

**Pull request merged**